### PR TITLE
Simple integration testing (adopted)

### PR DESCRIPTION
**Pull request by richchurcher**
# Objective

This older PR from `Wcubed` seemed well worth saving, adopted from #7314. See also tracking issue #2896 for ongoing discussion of Bevy testability. Thanks `Wcubed`!

## Solution

- Updated for 0.15
- Added the `expected`/`actual` pattern
- Switched to function plugin
- Tweaked a bit of description

## Testing

Green.

### CODE REVIEW

**Comment by Wcubed** [comment]
Thanks for adopting the pr 😃. I'd almost forgotten I made this one after two years 😄.

File: tests/how_to_test_apps.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#![allow(missing_docs)]
+/// Demonstrates simple integration testing of Bevy applications.
```
Comment (alice-i-cecile) [inline tests/how_to_test_apps.rs:2]: These should be `//!` comments, since they're at the start of the file.

File: tests/how_to_test_apps.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
+#![allow(missing_docs)]
```
Comment (alice-i-cecile) [inline tests/how_to_test_apps.rs:1]: I think this is only needed because your doc comments below are broken.

```suggestion
```

**Review by alice-i-cecile — COMMENTED** [review]
Good call: I do really like this! Straightforward and useful. There's a nit to be fixed about doc comments, but I'm happy to approve this after that.

**Review by BenjaminBrienen — APPROVED** [review]
This will be super useful!

File: tests/how_to_test_apps.rs (hunk 1 above)
Comment (richchurcher) [reply tests/how_to_test_apps.rs:2]: Argh, I had it that way and changed it. Have now read the rustdoc book in penance :laughing:

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**